Urban-Fun/View Controllers/TimelineViewController.m

```diff
@@ -48,30 +49,47 @@ - (void)viewDidLoad {
     if (CLLocationManager.locationServicesEnabled){
         [self.manager startUpdatingLocation];
     }
+    self.isMoreDataLoading = NO;
+    self.setsLoaded = 0;
+    [self getActivities];
 }
 
 - (void)beginRefresh:(UIRefreshControl *)refreshControl {
+    self.setsLoaded = 0;
     [self getActivities];
-    
-    
     [self.tableView reloadData];
-    
 }
 
+
+
 -(void)getActivities{
     PFQuery *activityQuery = [Activity query];
     [activityQuery orderByDescending:@"createdAt"];
     [activityQuery includeKey:@"host"];
     activityQuery.limit = 20;
+    activityQuery.skip = self.setsLoaded * 20;
+
     
   
     [activityQuery findObjectsInBackgroundWithBlock:^(NSArray<Activity *> * _Nullable activities, NSError * _Nullable error) {
-        if (activities) {
+        if (activities && self.setsLoaded > 0) {
+            NSMutableArray *temporaryActivitiesArray = [[NSMutableArray alloc] initWithArray:self.arrayOfActivities];
+            for (Activity *activity in activities){
+                if (![self.arrayOfActivities containsObject:activity]){
```

Comment:
Since you're loading a new set of activities, I think this will always be true, right? if so, I think you can use `addObjectsFromArray` instead, take a look to the [documentation](https://developer.apple.com/documentation/foundation/nsmutablearray/1408963-addobjectsfromarray)